src/context.rs

```diff
@@ -302,6 +306,20 @@ impl Context {
       };
     }
   }
+
+  /// Create a new context from a (non-default) context snapshot. There
+  /// is no way to provide a global object template since we do not create
+  /// a new global object from template, but we can reuse a global object.
+  pub fn from_snapshot<'s>(
+    scope: &mut HandleScope<'s, ()>,
```

Comment:
Do we need the scope here? Can this just take an `&mut Isolate`?

Reply:
In most cases you need to use `scope` immediately after calling this function, I think it's not worth the complexity.